openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,841 @@
+            nparams,params = parameters.shape
+            covsize = cov.shape[0]
+            mpar = int(covsize/nparams)
+            
+            #Add parameters from File 2
+            parameters = file2contributions(parameters, file2params)
+
+            # Create instance of MultiLevelBreitWignerCovariance
+            mlbw = cls(energy_min, energy_max)
+            mlbw.parameters = parameters
+            mlbw.covariance = cov
+            mlbw.mpar  = mpar
+            mlbw.lcomp = LCOMP
+
+            return mlbw
+
+        elif LCOMP == 0 :
+            cov = np.zeros([4,4])
+            records = []
+            cov_index = 0
+            for i in range(NLS):
+                items, values = get_list_record(file_obj)
+                num_res = items[5]
+                for j in range(num_res):
+                    one_res = values[18*j:18*(j+1)]
+                    res_values = one_res[:6]
+                    cov_values = one_res[6:]
+
+                    energy = res_values[0]
+                    spin = res_values[1]
+                    gt = res_values[2]
+                    gn = res_values[3]
+                    gg = res_values[4]
+                    gf = res_values[5]
+                    records.append([energy, spin, gt, gn, gg, gf])
+
+                    #Populate the coviariance matrix for this resonance
+                    #There are no covariances between resonances in LCOMP=0
+                    cov[cov_index,cov_index]=cov_values[0]
+                    cov[cov_index+1,cov_index+1:cov_index+2]=cov_values[1:2]
+                    cov[cov_index+1,cov_index+3]=cov_values[4]
+                    cov[cov_index+2,cov_index+2] = cov_values[3]
+                    cov[cov_index+2,cov_index+3] = cov_values[5]
+                    cov[cov_index+3,cov_index+3] = cov_values[6]
+
+                    cov_index += 4
+                    if j < num_res-1: #Pad matrix for additional values
+                        cov = np.pad(cov,((0,4),(0,4)),'constant',
+                                constant_values=0)
+
+
+            #Create pandas DataFrame with resonance data, currently 
+            #redundant with data.IncidentNeutron.resonance
+            columns = ['energy', 'J', 'totalWidth','neutronWidth',
+                       'captureWidth', 'fissionWidth']
+            parameters = pd.DataFrame.from_records(records, columns=columns)
+
+            #Determine mpar (number of parameters for each resonance in
+            #covariance matrix)
+            nparams,params = parameters.shape
+            covsize = cov.shape[0]
+            mpar = int(covsize/nparams)
+
+            #Add parameters from File 2
+            parameters = file2contributions(parameters, file2params)
+
+            # Create instance of class
+            mlbw = cls(energy_min, energy_max)
+            mlbw.parameters = parameters
+            mlbw.covariance = cov
+            mlbw.mpar  = mpar
+            mlbw.lcomp = LCOMP
```

Comment:
This block of code for creating the instance is the same on each branch of the `if`, so it can just be outside the `if` at the end. I would also change the signature of `__init__` to `__init__(self, energy_min, energy_max, parameters, covariance, mpar, lcomp)`.